Occupation dropdown backend

hrit2773 opened the pull request:
## Proposed Changes

- Changed the enum class

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

hrit2773 committed:
Occupation dropdown backend

hrit2773 committed:
Merge branch 'coronasafe:develop' into second-issue

hrit2773 committed:
Merge branch 'coronasafe:develop' into second-issue

rithviknishad reviewed (changes requested):
Missing migrations. Run `python manage.py makemigrations`

hrit2773 committed:
added migrations

rithviknishad commented on Makefile:
```diff
@@ -1,7 +1,7 @@
 
 
-DOCKER_VERSION := $(shell docker --version 2>/dev/null)
+DOCKER_VERSION := docker --version
```
Why was this changed?

hrit2773 committed:
fixed

hrit2773 commented:
@rithviknishad I changed this in my windows machine just to run the command and pushed it by mistake I changed it back and i guess its fine now added migration also

rithviknishad commented on care/facility/migrations/0426_alter_patientmetainfo_occupation.py:
```diff
@@ -0,0 +1,54 @@
+                    (8, "Business_related"),
+                    (9, "ENGINEER"),
+                    (10, "TEACHER"),
+                    (11, "Other_Professions"),
```
Do we need both "Other_Pressions" and "OTHERS"

hrit2773 replied:
@rithviknishad maybe we can consider other skilled professions that was mentioned in the dropdown suggstions list so i added

hrit2773 replied:
@rithviknishad in the frontend Other professional occupations were mentioned and professional occupation is a separate type and the corresponding value is OTHER_PROFESSIONS not to be confused with OTHERS and its not the same. This is needed.

rithviknishad replied:
Rename it in backend too. It could lead to confusions.

Also change to use uppercase for all

hrit2773 committed:
capitalized the names

hrit2773 commented:
@rithviknishad I capitalized the names and also capitalized in the frontend

hrit2773 committed:
changed back

hrit2773 committed:
changed to other professional occupations

hrit2773 committed:
changed other professional occupations

sainak commented:
@hrit2773 delete the migrations and rerun makemigrations

hrit2773 committed:
updated migrations

hrit2773 committed:
changed finally

hrit2773 commented:
@sainak @rithviknishad plz review this pr

rithviknishad commented on care/facility/models/patient.py:
```diff
@@ -570,7 +570,33 @@ class OccupationEnum(enum.Enum):
         HEALTH_CARE_LAB_WORKER = 4
         ANIMAL_HANDLER = 5
-        OTHERS = 6
+        HEALTHCARE_PRACTITIONER = 6
+        PARADEMICS = 7
+        BUSINESS_RELATED = 8
+        ENGINEER = 9
+        TEACHER = 10
+        OTHER_PROFESSIONAL_OCCUPATIONS = 11
+        OFFICE_ADMINISTRATIVE = 12
+        CHEF = 13
+        PROTECTIVE_SERVICE = 14
+        HOSPITALITY = 15
+        CUSTODIAL = 16
+        CUSTOMER_SERVICE = 17
+        SALES_SUPERVISOR = 18
+        RETAIL_SALES_WORKER = 19
+        INSURANCE_SALES_AGENT = 20
+        SALES_REPRESENTATIVE = 21
+        REAL_ESTATE = 22
+        CONSTRUCTION_EXTRACTION = 23
+        AGRI_NATURAL = 24
+        PRODUCTION_OCCUPATION = 25
+        PILOT_FLIGHT = 26
+        VEHICLE_DRIVER = 27
+        MILITARY = 28
+        HOMEMAKER = 29
+        OTHERS = 30
+        UNKNOWN = 31
+        NOT_APPLICABLE = 32
```
`OTHERS` was previously `6` and is now `30`. And `6` is now Healthcare Practitioner. This would mean, all previously entered patients with occupation as "Other" before this PR would become a Healthcare Practitioner once this is merged.

hrit2773 committed:
changed others

hrit2773 committed:
Merge branch 'develop' into second-issue

hrit2773 commented:
@rithviknishad yes you are right OTHERS would be misread as HEALTHCARE PRACTIOTIONER so changed it. And i made changes in the frontend also

hrit2773 committed:
Merge branch 'develop' into second-issue

nihal467 commented:
@hrit2773 there is migration issue, fix it

hrit2773 commented:
@nihal467 but it's working fine when I'm running it the occupation is being added and displayed properly. Should I re-run the migrations?

hrit2773 committed:
migration issue fix

hrit2773 committed:
changed

hrit2773 commented:
@nihal467 can you check it now maybe there were some conflicting migrations before

vigneshhari committed:
Merge branch 'develop' into second-issue
